### PR TITLE
Organizations updates

### PR DESCRIPTION
**Pull request by UnsungHero97**
- start using `Page`
- start using Epics
- pulling in changes from `feature/permissions` branch
- decorating organizations with `isOwner` property

something to focus on for this PR is how I do the organizations decoration. the epic fetches organizations and stores them in the redux store, then does the `checkAuthorizations()` call and decorates the organizations and updates the redux store with the decorated organizations.

I think this will be a common problem, that of decorating data via chained API calls. how can we do this better?

### CODE REVIEW

File: src/components/asynccontent/AsyncContent.js [hunk 1]
```diff
@@ -37,10 +39,7 @@ export default class AsyncContent extends React.Component {
 
   renderLoading() {
     return (
-      <div>
-        <h2>Loading</h2>
-        <FontAwesome name="spinner" size="5x" spin />
-      </div>
+      <LoadingSpinner />
```
Comment (mindblight) [inline src/components/asynccontent/AsyncContent.js]: +1. Let's move `LoadingSpinner` into `components/asynccontent`. I don't think it needs it's own directory

File: src/containers/catalog/CatalogComponent.js [hunk 2]
```diff
@@ -2,15 +2,14 @@ import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { denormalize } from 'normalizr';
 
-import Page from "../../components/page/Page";
+import Page from '../../components/page/Page';
```
Comment (mindblight) [inline src/containers/catalog/CatalogComponent.js]: +1

File: src/containers/organizations/components/OrganizationsComponent.js [hunk 3]
```diff
@@ -213,57 +178,83 @@ class OrganizationList extends React.Component {
...
           value={this.props.selectedOrgOption}
           onChange={updateSelectedOrganization} />
     );
   };
 
-  render() {
+  renderCreateOrganizationComponent = () => {
+
+    return (
+      <CreateOrganization
```
Comment (mindblight) [inline src/containers/organizations/components/OrganizationsComponent.js]: One side note: I've started making the `Create...` components smart components. It makes it a lot easier to move them around in the UI when redesigns happen, and keeping the async creation logic isolated to them is convenient.

Comment (UnsungHero97) [inline src/containers/organizations/components/OrganizationsComponent.js]: it is a smart component

Comment (mindblight) [inline src/containers/organizations/components/OrganizationsComponent.js]: +1

File: src/containers/organizations/epics/CreateOrganizationEpic.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    .catch((error) => {
+      console.error(error);
+    });
+}
+
+export default function createOrgEpic(action$ :Observable<Object>) {
+
+  return action$
+    .ofType(OrgsActionTypes.CREATE_NEW_ORG)
+    .mergeMap((action :Object) => {
```
Comment (mindblight) [inline src/containers/organizations/epics/CreateOrganizationEpic.js]: `.mergeMap(createNewOrg)`

File: src/containers/organizations/epics/CreateOrganizationEpic.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    .catch((e) => {
+      console.error(e);
+    });
+}
+
+function createNewOrg(action :Object) {
+
+  return Observable
+    .from(OrganizationsApi.createOrganization(action.org))
+    .mergeMap((newOrgId :UUID) => {
```
Comment (mindblight) [inline src/containers/organizations/epics/CreateOrganizationEpic.js]: `.mergeMap(setNewOrgPermissions)`

File: src/containers/organizations/epics/CreateOrganizationEpic.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      return setNewOrgPermissions(newOrgId);
+    })
+    .catch((error) => {
+      console.error(error);
```
Comment (mindblight) [inline src/containers/organizations/epics/CreateOrganizationEpic.js]: Eventually, you'll want to return an error action for the reducer to pick up and display to the user

File: src/containers/organizations/epics/CreateOrganizationEpic.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  PrincipalBuilder
+} = DataModels;
+
+const {
+  ActionTypes,
+  PermissionTypes,
+  PrincipalTypes
+} = Types;
+
+function setNewOrgPermissions(newOrgId :UUID) {
```
Comment (mindblight) [inline src/containers/organizations/epics/CreateOrganizationEpic.js]: This should be moved into `constainers/permissions/PermissionEpic.js` (doesn't exist yet, feel free to create) as it's own epic: 
```
updatePermissionsEpic(actions$) {
  actions$.ofType(actionTypes.UPDATE_ACL)
  ...
}
```

and `createNewOrg` should put an `UPDATE_ACL` request in the observable. See `CatalogEpic.searchCatalog` for an example

File: src/containers/organizations/epics/CreateOrganizationEpic.js (hunk 4 above)
Comment (UnsungHero97) [inline src/containers/organizations/epics/CreateOrganizationEpic.js]: doh

File: src/containers/organizations/epics/CreateOrganizationEpic.js (hunk 6 above)
Comment (UnsungHero97) [inline src/containers/organizations/epics/CreateOrganizationEpic.js]: +1

**Pull request merged**